ArDOS/ArdosParser/DefaultParser.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Drawing;
+using System.Text.RegularExpressions;
+using ArDOS.Model;
+using ArDOS.Parser.Exceptions;
+using ArDOS.Parser.Actions;
+
+namespace ArDOS.Parser
+{
+    public partial class WindowsParser
+    {
+        protected const RegexOptions DEFAULT_RE_OPTS = RegexOptions.Multiline | RegexOptions.Compiled | RegexOptions.IgnorePatternWhitespace;
+        public static ArdosMenu Parse(string input)
+        {
+            var sections = Regex.Split(input, @"\n^-*$\n", DEFAULT_RE_OPTS).Select(x => x.Split(new [] { "\n", "\r\n" }, StringSplitOptions.RemoveEmptyEntries)).ToArray();
+
+            // Parse title
+            var titleSection = sections[0];
+            var titleItem = ParseItem(sections[0][0], 0);
+
+            // Parse sections
+            var contentSections = sections[1..];
+            var items = ParseSections(contentSections);
+
+            return new ArdosMenu(titleItem.Text, titleItem.Image, items);
+        }
+
+        protected static ArdosItem[][] ParseSections(string[][] lines)
+        {
+            return lines.Select(ParseSection).ToArray();
+        }
+
+        protected static ArdosItem[][] ParseSections(string input)
+        {
+            return ParseSections(Regex.Split(input, @"\n^-*$\n", DEFAULT_RE_OPTS).Select(x => x.Split(new[] { "\n", "\r\n" }, StringSplitOptions.RemoveEmptyEntries)).ToArray());
+        }
+
+        protected static ArdosItem[] ParseSection(string[] lines)
+        {
+            var items = new List<ArdosItem>();
+            for (int i = 0; i < lines.Length; i++)
+            {
+                if (lines[i].TrimStart().StartsWith("--"))
```

Comment:
What about submenus with alternative items?

Reply:
I handle them, this `if` is for recursively handling submenus